grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaChannel.java

```diff
@@ -107,6 +105,10 @@ public <I, O> ClientCall<I, O> newCall(
         final boolean unsafeWrapResponseBuffers = options.get(GrpcClientOptions.UNSAFE_WRAP_RESPONSE_BUFFERS);
         final Compressor compressor = options.get(GrpcClientOptions.COMPRESSOR);
         final DecompressorRegistry decompressorRegistry = options.get(GrpcClientOptions.DECOMPRESSOR_REGISTRY);
+        final Set<String> availableEncodings = decompressorRegistry.getAdvertisedMessageEncodings();
+        if (!availableEncodings.isEmpty()) {
+            advertisedEncodingsHeader = String.join(",", availableEncodings);
+        }
```

Comment:
It looks like we already pass `decompressorRegistry` when creating an `ArmeriaClientCall` and `advertisedEncodingHeader` is used only once in `ArmeriaClientCall.prepareHeaders()`. So, how about optimizing like this?

- Remove this block completely as well as `ArmeriaChannel.advertisedEncodingHeader`
- Do not pass `availableEncodings` when creating an `ArmeriaClientCall`.
- Use `DecompressorRegistry.getRawAdvertisedMessageEncodings()` to add the header in `ArmeriaClientCall.prepareHeaders()`.